tiledb/sm/misc/resource_pool.h

```diff
@@ -139,6 +144,94 @@ class ResourcePool {
   friend class ResourceHandle<T, tiledb::sm::ResourcePool>;
 };
 
+template <class T>
+class BlockingResourcePool {
+ public:
+  using resource_handle = ResourceHandle<T, tiledb::sm::BlockingResourcePool>;
+
+  /** Constructor. */
+  BlockingResourcePool(unsigned int n)
+      : resources_(n)
+      , unused_(n)
+      , unused_idx_(n - 1)
+      , num_blocked_threads_(0) {
+    for (unsigned int i = 0; i < n; i++)
+      unused_[i] = i;
+  }
+
+  /** Take a resource from the pool. */
+  resource_handle take() {
```

Comment:
This is great to know, thanks!
For others that might be interested, looking it up to understand why this can happen, I saw this is called [[spurious wakeup](url)](https://en.wikipedia.org/wiki/Spurious_wakeup)